lnst/RecipeCommon/Perf/Measurements/LinuxPerfMeasurement.py

```diff
@@ -58,6 +55,10 @@ def __init__(
             except FileExistsError:
                 pass
 
+    @property
+    def hosts(self) -> list[Host]:
+        return self.linuxperf_cpus.keys()
```

Comment:
I think return type is incorrect:
```python
>>> isinstance({}.keys(), list)
False
```

Reply:
Fixed.

Reply:
I'd rather see this returning an `Iterable[Host]` and keeping the previous version. Is there a use-case where you need this to be a list? E.g. you need to index it or append to it, or do we just always iterate over it?

Reply:
At the moment only iteration over it is sufficient.

Reply:
Took another look, we need it to be list: https://github.com/LNST-project/lnst/blob/10129c50d852d944ceab42c1d8bfc011763d94cc/lnst/RecipeCommon/Perf/Measurements/LinuxPerfMeasurement.py#L83

Reply:
Nice catch. Sure, we can keep this a list. In that case though doing `list(self.linuxperf_cpus.keys())` is sufficient.

Or we could change the version getting to `next(self.hosts)` and raise if `StopIteration` happens. I'll leave this decision for you.

Reply:
I'd like to keep Iterable since it makes more sense - we just need it for reading the value, not changing it. Thanks for your proposal, makes sense.

Reply:
Would it make sense to change `hosts` property to return `iter(self.linuxperf_cpus.keys())`? And change the return type hint to `Iterator[Host]`?

Reply:
I can't, because iterator has no `len()` that is called in:
```python
    def _get_host_perf_version(self) -> Optional[str]:                          
        if not len(self.hosts):                                                                                                                                               
```

```python
>>> d = {'a': 1, 'b': 2}
>>> ks = d.keys()
>>> i1 = iter(ks)
>>> len(i1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: object of type 'dict_keyiterator' has no len()
>>> len(ks)
2
```

Reply:
@Kuba314 I update the patch, could you take a look?

Reply:
> > Took another look, we need it to be list
> 
> So isn't `list(...keys())` the easiest way?

Well, it most likely is. The problem I have with that is that we're unnecesarilly creating a list where it's never really used as a list (except the unusual fetch of first item).

Also, easier in what way? I only had to add one additional line to the code compared to the version I used list.

Reply:
the actual list creation is not really that slow / memory intensive. But...

We are not guaranteed that there will be at least one host, right? I'm either for changing the return value to an `Iterator` (`yield from .keys()` or `iter(.keys())`) or just returning the list (`list(.keys())`.

The version checking will then use either `next()` or `[0]`, but both should be fine, because there will always be at least one host, right?

Reply:
I think the `hosts` property should always return the full content, by using `yield`, this would not work.

I think converting to list makes probably the most sense at this point as it should be most prone to any side effects.

Reply:
Should be ready now.